### PR TITLE
Chore: Gradle plugin cleanup

### PR DESCRIPTION
**Pull request by mvysny**
## Description

Does not fix any particular issue. Contains a couple of cleanups of the Vaadin Gradle Plugin.

## Type of change

- [ ] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPrepareFrontendTask.kt [hunk 1]
```diff
@@ -89,13 +88,6 @@ public open class VaadinPrepareFrontendTask : DefaultTask() {
         val adapter = GradlePluginAdapter(project, true)
         val tokenFile = BuildFrontendUtil.propagateBuildInfo(adapter)
 
-        if (extension.requireHomeNodeExec) {
```
Comment (czp13) [inline flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPrepareFrontendTask.kt:92]: So this issue is not happening anymore so we can delete and also the `createFrontendTools()` method as well below?

**Review by czp13 — APPROVED** [review]
This seems safe to me, only minor question about the older issue:
https://github.com/vaadin/vaadin-gradle-plugin/issues/76
which had a fixing PR around 2020, so maybe/probably it is fixed, just to double-check.

Thanks for the work @mvysny!

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPrepareFrontendTask.kt (hunk 1 above)
Comment (mvysny) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPrepareFrontendTask.kt:92]: Yup, exactly. This code was actually moved inside of the BuildFrontendTool.prepareFrontend() function some time ago.

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinUtils.kt [hunk 2]
```diff
@@ -61,20 +63,17 @@ public fun Project.getBooleanProperty(propertyName: String) : Boolean? {
...
-    return FrontendTools(settings)
-}
-
 /**
  * Returns only jar files from given file collection.
  */
 internal val Configuration.jars: FileCollection
     get() = filter { it.name.endsWith(".jar", true) }
+
+internal val Project.sourceSets: SourceSetContainer get() = project.properties["sourceSets"] as SourceSetContainer
```
[the remark this answers is not in the document]
Comment (mvysny) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinUtils.kt:77]: Yup, just moving utility functions to their proper place.

**Comment by mvysny** [comment]
Thank you kindly for the review 🙇 the referenced issue remains fixed since the code was moved as noted above.

since all items have been resolved, I’ll merge the pr

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPrepareFrontendTask.kt (hunk 1 above)
Comment (czp13) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinPrepareFrontendTask.kt:92]: Thank you, makes sense :)!

File: flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinUtils.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (czp13) [reply flow-plugins/flow-gradle-plugin/src/main/kotlin/com/vaadin/gradle/VaadinUtils.kt:77]: Thank you! 🙇

**Pull request merged**